components/style/stylesheets/supports_rule.rs

```diff
@@ -125,21 +125,18 @@ impl SupportsCondition {
         // Whitespace is normally taken care of in `Parser::next`,
         // but we want to not include it in `pos` for the SupportsCondition::FutureSyntax cases.
         while input.try(Parser::expect_whitespace).is_ok() {}
-        let pos = input.position();
         match input.next()? {
             Token::ParenthesisBlock => {
                 input.parse_nested_block(|input| {
-                    // `input.try()` not needed here since the alternative uses `consume_all()`.
-                    parse_condition_or_declaration(input).or_else(|_| {
-                        consume_all(input);
-                        Ok(SupportsCondition::FutureSyntax(input.slice_from(pos).to_owned()))
+                    input.try(|i| parse_condition_or_declaration(i)).or_else(|_| {
+                        let content = parse_any_value(input)?;
+                        Ok(SupportsCondition::FutureSyntax(format!("({})", content)))
                     })
                 })
             }
-            Token::Function(_) => {
-                let result: Result<_, ParseError> = input.parse_nested_block(|i| Ok(consume_all(i)));
-                result.unwrap();
-                Ok(SupportsCondition::FutureSyntax(input.slice_from(pos).to_owned()))
+            Token::Function(name) => {
+                let content = input.parse_nested_block(|i| parse_any_value(i))?;
+                Ok(SupportsCondition::FutureSyntax(format!("{}({})", name, content)))
```

Comment:
hmm, I thought it would be useful to have a `parse_any_value` returning the value parsed... but yeah, probably in this case it is better to just keep using `slice_from` in the callsites.